--- glib-1.2.10/configure.in.0010~	2013-01-15 22:05:20.359362282 +0100
+++ glib-1.2.10/configure.in	2013-01-15 22:05:45.970055270 +0100
@@ -7,7 +7,7 @@ builtin(include, acglib.m4)dnl
 AC_PREREQ(2.54)
 
 # init autoconf (and check for presence of glist.c)
-AC_INIT(glist.c)
+AC_INIT(glib, 1.2.10)
 
 # Save this value here, since automake will set cflags later
 cflags_set=${CFLAGS+set}
@@ -57,10 +57,10 @@ AC_SUBST(LT_AGE)
 VERSION=$GLIB_VERSION
 PACKAGE=glib
 
-AM_INIT_AUTOMAKE($PACKAGE, $VERSION, no-define)
+AM_INIT_AUTOMAKE
 
 # Specify a configuration file
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_DEFINE_UNQUOTED(GLIB_MAJOR_VERSION, $GLIB_MAJOR_VERSION)
 AC_DEFINE_UNQUOTED(GLIB_MINOR_VERSION, $GLIB_MINOR_VERSION)
@@ -150,7 +150,6 @@ AC_DEFINE_UNQUOTED(G_COMPILED_WITH_DEBUG
 
 # Checks for programs.
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_PROG_INSTALL
 
 changequote(,)dnl
